internal/kms/kms.go

```diff
@@ -368,7 +368,7 @@ type DEKStore interface {
 	// FetchDEK reads the DEK from the configured store and returns it.
 	FetchDEK(volumeID string) (string, error)
 	// RemoveDEK deletes the DEK from the configured store.
-	RemoveDEK(volumeID string) error
+	RemoveDEK(ctx context.Context, volumeID string) error
```

Comment:
please add the context for the other DEKStore functions too.

Reply:
This will require to add context for `EncryptDEK`,  `DecryptDEK`  and `GetSecret` funcs as we are calling `FetchDEK` in secretskms.go

https://github.com/ceph/ceph-csi/blob/328e4e5a0fa06521c5508b94a5b7ff250ada1681/internal/kms/secretskms.go#L209-L213

IMO, it will be good to make these changes in another PR?

Reply:
And requires changes at lot of places and will make the PR too big.
Lets get this PR in and I'll make the context change in follow up PR.

Reply:
follow up PR: https://github.com/ceph/ceph-csi/pull/4477